recipes-openxt/openxt/openxt-measuredlaunch/ml-functions

```diff
@@ -0,0 +1,375 @@
+#!/bin/sh
+#
+# Copyright (c) 2016 Daniel P. Smith, Apertus Solutions, LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+
+. /usr/lib/openxt/key-functions
+[ $? -eq 0 ] || {
+    echo "failed to load key-functions"
+    exit 1
+}
+
+hex2bin() {
+    read hex
+    echo -n $hex | sed 's/\([0-9A-F]\{2\}\)/\\\\\\x\1/gI' | xargs printf
+}
+
+# Function to parse grub.cfg and return all multiboot modules and their
+#   full command line
+# parameter 0: path to grub.conf
+parse_grub() {
```

Comment:
Aren't we trying to do tboot's work here? Can't we just do a `txt-stat`? Parsing the grub config file really doesn't seem ideal...